packages/react-composites/tests/app/lib/useFakeChatAdapters.ts

```diff
@@ -0,0 +1,161 @@
+
+import { useEffect, useState } from 'react';
+import { ChatClient, ChatParticipant, ChatThreadClient } from '@azure/communication-chat';
+import { CommunicationTokenCredential, CommunicationUserIdentifier } from '@azure/communication-common';
+import { CommunicationIdentifier } from '@azure/communication-signaling';
+import { _createStatefulChatClientWithDeps } from '@internal/chat-stateful-client';
+import { RestError } from '@azure/core-http';
+
+/**
+ * Fake adapters and data structures for in-memory fake-backend for chat.
+ */
+export interface FakeAdapters {
+  local: ChatAdapter;
+  remotes: ChatAdapter[];
+  service: {
+    model: Model;
+    threadId: string;
+  };
+}
+
+/**
+ * Create chat adapters using an in-memory fake-backend for chat.
+ */
+export function useFakeChatAdapters(args: FakeChatAdapterArgs): FakeAdapters | undefined {
+  const [fakeAdapters, setFakeAdapters] = useState<FakeAdapters>();
+  useEffect(() => {
+    (async (): Promise<void> => {
+      if (!args.localParticipant.displayName) {
+        throw new Error(
+          `Local participant must have display name defined, got ${JSON.stringify(args.localParticipant)}`
+        );
+      }
+
+      const chatClientModel = new Model({ asyncDelivery: false });
+      const participants = orderParticipants(
+        args.localParticipant,
+        args.remoteParticipants,
+        args.localParticipantPosition
+      );
+      const chatClient = new FakeChatClient(chatClientModel, args.localParticipant.id);
+      const thread = await chatClient.createChatThread({ topic: 'Cowabunga' }, { participants });
+      const threadId = thread?.chatThread?.id ?? '';
+      const chatThreadClient = chatClient.getChatThreadClient(threadId);
+      const adapter = await initializeAdapter(
+        {
+          userId: args.localParticipant.id,
+          displayName: args.localParticipant.displayName,
+          chatClient: chatClient as IChatClient as ChatClient,
+          chatThreadClient: chatThreadClient
+        },
+        args.chatThreadClientMethodErrors
+      );
+      const newFakeAdapters: FakeAdapters = {
+        local: adapter,
+        remotes: [],
+        service: {
+          model: chatClientModel,
+          threadId: threadId
+        }
+      };
+      if (args.participantsWithHiddenComposites) {
+        newFakeAdapters.remotes = await initializeAdapters(
+          args.participantsWithHiddenComposites,
+          chatClientModel,
+          thread
+        );
+      }
+      setFakeAdapters(newFakeAdapters);
+    })();
+  }, [args]);
+
+  return fakeAdapters;
+}
+
+const initializeAdapters = async (
+  participants: ChatParticipant[],
+  chatClientModel: Model,
+  thread
+): Promise<ChatAdapter[]> => {
+  const remoteAdapters: ChatAdapter[] = [];
+  for (const participant of participants) {
+    if (!participant.displayName) {
+      throw new Error(`All participants must have displayName defined, got ${JSON.stringify(participant)}`);
```

Comment:
nit: we should use a safe json stringingify to prevent the error not being displayed if JSON stringify throws an error

Reply:
While I agree, I don't want to introduce a dependency from `@internal/acs-ui-common` in this PR.

Reply:
Will try to remember and follow up on top of the stack.